addons/udes_sale_stock/models/stock_move.py

```diff
@@ -48,3 +48,24 @@ def _prepare_procurement_values(self):
         values = super()._prepare_procurement_values()
         values["sale_line_id"] = self.sale_line_id.id
         return values
+    
+    def _search_picking_for_assignation(self):
+        """
+        Override the method to filter and retrieve pickings based on the priority
+        of the associated sale order.
+        """
+        pickings = super(StockMove, self)._search_picking_for_assignation()
+        priority = self.sale_line_id.order_id.priority
```

Comment:
Sorry just noticed something, please can this be:
```suggestion
        if self.sale_line_id:
               priority = self.sale_line_id.order_id.priority
               pickings.filtered(lambda p: p.priority = priority)
        return pickings
```
As this might get called when the picking is not created from a sales order.

Likewise for `_get_new_picking_values()`:
```suggestion
        if self.sale_line_id:
               values.update({'priority': self.sale_line_id.order_id.priority})
        return values
```